types/sphere-engine-browser/sphere1-api.d.ts

```diff
@@ -0,0 +1,1380 @@
+}
+
+interface SphereSpritesetBase {
+    x1: number;
+    y1: number;
+    x2: number;
+    y2: number;
+}
+
+interface SphereSpritesetDirection {
+    /** Name of the direction */
+    name: string;
+    /** Array of spriteset frame objects */
+    frames: SphereSpritesetFrame[];
+}
+
+interface SphereSpritesetFrame {
+    /** Index into images array */
+    index: number;
+    /** Number of frames before animation should switch */
+    delay: number;
+}
+
+interface SphereWindowStyle {
+    /**
+     * Draws the window at `x`,`y` with the width and height of `w` and `h`.
+     * Note that window corners and edges are drawn outside of the width
+     * and height of the window.
+     */
+    drawWindow(x: number, y: number, w: number, h: number): void;
+    /**
+     * Sets the color mask for a windowstyle to `color`
+     * @see ApplyColorMask
+     */
+    setColorMask(color: Sphere1Color): void;
+    /**
+     * Gets the color mask being used by the windowstyle object
+     */
+    getColorMask(): Sphere1Color;
+    /**
+     * Return the border size (height or width) in pixels for the windowstyle object.
+     * You can use EDGE_LEFT, EDGE_TOP, EDGE_RIGHT and EDGE_BOTTOM as parameters.
+     * Note that the corner sizes are ignored.
+     */
+    getBorder(border: number): number;
+}
+
+/*
+ * Sphere 1.x functions
+ */
+
+/**
+ * Returns the Sphere API version (2 for miniSphere)
+ */
+declare function GetVersion(): number;
+
+/**
+ * Returns the API version string (v2.0 for miniSphere)
+ */
+declare function GetVersionString(): string;
+
+/**
+ * Invoke the garbage collector (you shouldn't need to use this in most cases)
+ */
+declare function GarbageCollect(): void;
+
+/**
+ * Returns an array of object literals with properties describing the games in
+ * `$HOME/Documents/miniSphere/Games/` for creating a startup game loader
+ */
+declare function GetGameList(): SphereGame[];
+
+/**
+ * Runs a Sphere game, given its directory
+ */
+declare function ExecuteGame(directory: string): void;
+
+/**
+ * Restarts the current game (use Sphere.restart() instead)
+ * @deprecated
+ */
+declare function RestartGame(): void;
+
+/**
+ * Exits miniSphere (use Sphere.shutDown() instead)
+ * @deprecated
+ */
+declare function Exit(): void;
+
+/**
+ * Displays the given message and exits (use Sphere.abort() instead)
+ * @deprecated
+ */
+declare function Abort(message: any): void;
+
+/**
+ * Opens `filename` as a log file
+ */
+declare function OpenLog(
+    filename: string,
+): {
+    /**
+     * writes `text` to the log file
+     */
+    write(text: string): void;
+    /**
+     * Begins an indented block in the text file with `name` as the block title
+     */
+    beginBlock(name: string): void;
+    /**
+     * Closes the current block
+     */
+    endBlock(): void;
+};
+
+/**
+ * Returns the number of milliseconds since some arbitrary time
+ * @example
+ * let start = GetTime();
+ * while (GetTime() < start + 1000) {
+ * 	// do stuff
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/sphere-engine-browser/sphere1-api.d.ts;
  Line: 575;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.